src/go/k8s/controllers/redpanda/console_controller.go

```diff
@@ -275,3 +286,33 @@ func (r *ConsoleReconciler) WithClusterDomain(
 	r.clusterDomain = clusterDomain
 	return r
 }
+
+func (r *ConsoleReconciler) reconcileConsoleForCluster(c client.Object) []reconcile.Request {
+	// Since Console is a managed Object, list requests should be handled at
+	// the level of client cache, so no real request to the cluster API.
+	// We set a strict timeout for this reason.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	cs := redpandav1alpha1.ConsoleList{}
+	err := r.Client.List(ctx, &cs) // all namespaces
+	if err != nil {
+		r.Log.Error(err, "unexpected: could not list consoles for propagating reconcile events")
+		return nil
+	}
+
+	var res []reconcile.Request
+	for i := range cs.Items {
+		cns := &cs.Items[i]
+		if cns.Spec.ClusterRef.Namespace == c.GetNamespace() && cns.Spec.ClusterRef.Name == c.GetName() {
```

Comment:
Wdyt about indexing with `clusterRef` and filtering with `MatchingFields`?

Reply:
Do you have an example? It will still be client-side filtering, right?

Reply:
yup something like [this](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.2/pkg/client#example-FieldIndexer-SecretName) it's client-side caching implemented by client-go, i think this way we don't need to list and query all consoles everytime there is a cluster change :pray:

Reply:
Under the hood you will always query all consoles every time. The most important thing is that indexer is client-side filtering.

Reply:
Yeah, it might be cleaner from a code pov, but I don't think it will add performance gains. Especially considering that a k8s cluster is expected to have 1 or a few Consoles at most.